### ISSUE TITLE
Investigate using LibXML

### ISSUE THREAD

**Issue by phuedx** [joakin/loot#9]
64e7a81 benchmarks the trumpet-based streaming transform against an _almost_ equivalent LibXML-based transform.

Here's the output on my mid 2013 MacBook Air (1.7 GHz Intel Core i7, 8 GB 1600 MHz DDR3):

> $ node bench/bench.js
> trumpet x 0.57 ops/sec ±2.00% (39 runs sampled)
> libxml x 9.24 ops/sec ±1.35% (524 runs sampled)

As noted above, the benchmarks are for the transforms in isolation against a single fixture. The fixture is read – streamed in the case of the trumpet-based transform – and transformed.

While the preliminary results suggest that using LibXML would "just work", we should also measure the following before making a decision:
- Time To First Byte
- Time until the `DOMContentLoaded` event is fired
- Memory consumption
- ...

**Comment by phuedx** [comment]
[phantomas](https://github.com/macbre/phantomas) might be of use here.

**Comment by joakin** [comment]
Should we benchmark also with domino? (the library that mobileapps use?)

We should pick one for now and run with it, do you think it's going to impact the output of the research?

**Comment by phuedx** [comment]
> Should we benchmark also with domino? (the library that mobileapps use?)

Probably. My gut feel would be that it's easier to develop with but slower than LibXML.

> We should pick one for now and run with it, do you think it's going to impact the output of the research?

Picking one and running with it will get us to where we need to be: an end-to-end demo with a reasonable idea about where to go next.

It's good to have options for improving the user-perceived performance of the server.

**Comment by joakin** [comment]
On my crapbook air:

```
trumpet x 0.61 ops/sec ±1.95% (41 runs sampled)
libxml x 8.68 ops/sec ±1.30% (497 runs sampled)
```

I say let's run with libxmljs for the moment.

**Comment by phuedx** [comment]
👍

After a little investigation, I found that we'd have to use [html-tokenize](https://github.com/substack/html-tokenize) and [html-select](https://github.com/substack/html-select) to fix #2 (and break the document up into sections).

**Comment by joakin** [comment]
See https://github.com/joakin/loot/pull/10
